apps/examples/testcase/le_tc/kernel/tc_libc_math.c

```diff
@@ -38,6 +38,875 @@
 #define VAL2 4503599627370496.5
 
 /**
+ * @fn                  :tc_libc_math_acos
+ * @brief               :Returns the principal value of the arc cosine of x
+ * @scenario            :Returns the principal value of the arc cosine of x
+ * @API's covered       :acos
+ * @Preconditions       :None
+ * @Postconditions      :None
+ * @Return              :void
+ */
+static void tc_libc_math_acos(void)
+{
+	const double in_val[5] = { 0.64, -0.4, ZERO, -1, 1 };
+	const double sol_val[5] = { 0.87629806116834, 1.9823131728624, 1.5707963267949, 3.1415926535898, 0 };
+	double ret_val[5];
+	int acos_idx;
+
+	/* Returns the principal value of the arc cosine of x, expressed in radians */
+
+	for (acos_idx = 0; acos_idx < 5; acos_idx++) {
+		ret_val[acos_idx] = acos(in_val[acos_idx]);
+		TC_ASSERT_LEQ("acos", fabs(sol_val[acos_idx] - ret_val[acos_idx]), FLT_EPSILON);
+	}
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+ * @fn                  :tc_libc_math_acosf
+ * @brief               :Returns the principal value of the arc cosine of x
+ * @scenario            :Returns the principal value of the arc cosine of x
+ * @API's covered       :acosf
+ * @Preconditions       :None
+ * @Postconditions      :None
+ * @Return              :void
+ */
+static void tc_libc_math_acosf(void)
+{
+	const float in_val[5] = { 0.64, -0.4, 0, -1, 1 };
+	const double sol_val[5] = { 0.8762981295586, 1.9823131561279, 1.5707963705063, 3.1412079334259, 0.0004112283350 };
+	float ret_val[5];
+	int acosf_idx;
+
+	/* Returns the nonnegative area hyperbolic cosine of x */
+
+	for (acosf_idx = 0; acosf_idx < 5; acosf_idx++) {
+		ret_val[acosf_idx] = acosf(in_val[acosf_idx]);
+		TC_ASSERT_LEQ("acosf", fabs(sol_val[acosf_idx] - ret_val[acosf_idx]), FLT_EPSILON);
```

Comment:
you should use **fabsf()** right, as you are checking API of float type, **acosf()**.